src/components/GoodsSearchProvider.php

```diff
@@ -0,0 +1,413 @@
+
+    /**
+     * @var string the material key
+     */
+    public $materialKey = 'goodsSearchMaterial';
+
+    /**
+     * @var string the current action route
+     */
+    public $actionRoute = ['/store/goods/search'];
+
+    /**
+     * @var string the block key
+     */
+    public $blockKey = 'goods';
+
+    /**
+     * @var string the pagination block key
+     */
+    public $paginationKey = 'pagination';
+
+    /**
+     * @var string the pagination block key
+     */
+    public $totalCountKey = 'totalCount';
+
+    /**
+     * @var string the categories block key
+     */
+    public $categoriesKey = 'categories';
+
+    /**
+     * @var string the image property key
+     */
+    public $imagePropertyKey = '';
+
+    /**
+     * @var int thumb width
+     */
+    public $thumbWidth = 262;
+
+    /**
+     * @var int thump height
+     */
+    public $thumbHeight = 175;
+
+    /**
+     * @var int goods per page
+     */
+    public $maxLimit = 12;
+
+
+    /**
+     * @var bool|int parent id
+     */
+    public $parentId = false;
+
+    /**
+     * @var string the pagination get parameter
+     */
+    public $paginationParameter = 'page';
+
+    /**
+     * @var string the search get parameter
+     */
+    public $searchParameter = 'search';
+
+    /**
+     * @var string the limit get parameter
+     */
+    public $limitParameter = 'limit';
+
+    /**
+     * @var string the limit get parameter
+     */
+    public $parentIdParameter = 'parent-id';
+
+    /**
+     * @var string the sort get parameter
+     */
+    public $sortParameter = 'sort';
+
+    /**
+     * @var array sort fields
+     */
+    public $sortFields = ['retail_price', 'id'];
+
+
+    public function pack()
+    {
+        return [
+            'class' => static::class,
+            'entities' => $this->entities,
+        ];
+    }
+
+    private function buildGood($good)
+    {
+        $language_id = \Yii::$app->multilingual->language_id;
+        $currency = CurrencyHelper::getUserCurrency();
+
+        if (!empty($this->imagePropertyKey)) {
+            $imageProperty = Property::findOne(['key' => $this->imagePropertyKey]);
+            $images = MediaHelper::getMediaData(
+                $good,
+                $imageProperty,
+                new GlideConfigurator($this->thumbWidth, $this->thumbHeight)
+            );
+            $image = reset($images);
+        }
+
+        ProductPrice::create($good);
+        $price = $good->getMinPrice(Price::TYPE_RETAIL, true, $currency->iso_code);
+
+        $properties = [];
+
+        foreach ($good->propertiesValues as $property_id => $property_value) {
+            if (isset($property_value[$language_id])) {
+                $properties[$good->propertiesAttributes[$property_id]] = $property_value[$language_id][0];
+            } else {
+                $properties[$good->propertiesAttributes[$property_id]] = $property_value;
+            }
+        }
+
+        $parent = $good->getMainCategory()->one();
```

Comment:
`getMainCategory()` is a relation method. You can call `mainCategory` without `one()` method